src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -248,23 +249,26 @@ private void EnqueueUpdate(ProjectId projectId)
         {
             // A race is not possible here because we use the main thread to synchronize the updates
             // by capturing the sync context.
-            if (!_deferredUpdates.TryGetValue(projectId, out var update) || update.IsCompleted)
+            if (_deferredUpdates.TryGetValue(projectId, out var deferredUpdate) && !deferredUpdate.Task.IsCompleted && !deferredUpdate.Task.IsCanceled)
             {
-                _deferredUpdates[projectId] = UpdateAfterDelay(projectId);
+                deferredUpdate.Cts.Cancel();
```

Comment:
In theory, we could keep calling for a deferred update repeatedly, and consequently an update is deferred indefinitely.

This shouldn't be an issue, as we use similar logic later on in the stack:
https://github.com/dotnet/aspnetcore-tooling/pull/1936/files#diff-8d2f7fc40b9c4a7c4aeb497a628bca44R73-R78

and consequently, such an issue was always there in theory. We should keep an eye out to see if we encounter any irregularities rooting from this issue over the next little bit.